### PR TITLE
Add binaries to enable running Fakes in Net Core

### PR DESCRIPTION
**Pull request by vritant24**
## Description
The .NetCore implementation for Fakes requires the CLRIE instrumentation engine, which is different from the .NET Framework implementation that uses Intellitrace. 
This PR adds the necessary binaries that allow for Fakes tests to run through the TestPlatform package alone. The internal package used to provide the Fakes binaries has also been renamed and will constitute of all binaries that will be shared between Fakes in VS and TestPlatform.

### CODE REVIEW

**Review by nohwnd — APPROVED** [review]

**Comment by nohwnd** [comment]
Looks good to me, we merge and insert into VS asap to see if we break stuff.

File: src/package/nuspec/Microsoft.TestPlatform.nuspec [hunk 1]
```diff
@@ -489,6 +489,17 @@
...
+    <file src="$FakesPackageDir$\TestPlatform\Microsoft.VisualStudio.TestPlatform.Fakes.dll"	target="tools\net451\Common7\IDE\Extensions\TestPlatform\Microsoft.VisualStudio.TestPlatform.Fakes.dll" />
+
+    <file src="$FakesPackageDir$\DataCollector\Microsoft.VisualStudio.Fakes.DataCollector.dll"	target="tools\net451\Common7\IDE\Extensions\TestPlatform\Extensions\Microsoft.VisualStudio.Fakes.DataCollector.dll" />
+    <file src="$FakesPackageDir$\DataCollector\Microsoft.Cci.dll"	target="tools\net451\Common7\IDE\Extensions\TestPlatform\Extensions\Microsoft.Cci.dll" />
+
+    <file src="$FakesPackageDir$\Instrumentation\x86\Microsoft.QualityTools.Testing.Fakes.Instrumentation.dll"	target="tools\net451\Common7\IDE\Extensions\TestPlatform\Extensions\Fakes\x86\Microsoft.QualityTools.Testing.Fakes.Instrumentation.dll" />
+    <file src="$FakesPackageDir$\Instrumentation\x86\FakesInstrumentationProfiler_x86.config"	target="tools\net451\Common7\IDE\Extensions\TestPlatform\Extensions\Fakes\x86\FakesInstrumentationProfiler_x64.config" />
```
Comment (AbhitejJohn) [inline src/package/nuspec/Microsoft.TestPlatform.nuspec:498]: The config linked does not seem right. This should be the x86 config...

**Review by AbhitejJohn — APPROVED** [review]
Please validate Fakes runtime scenarios for x86 and x64 for the xcopyable package. Also, we would want to validate VS workflows given we are building against the recent Fakes version.

File: src/package/nuspec/Microsoft.TestPlatform.nuspec (hunk 1 above)
Comment (vritant24) [reply src/package/nuspec/Microsoft.TestPlatform.nuspec]: Fixed. Validated x64, x86 for netcoreapp3.1 and net472

**Comment by vritant24** [comment]
Validated VS scenario as well for net472 + netcoreapp3.1

**Pull request merged**